Applications/Spire/Source/Ui/Slider2D.cpp

```diff
@@ -253,6 +253,24 @@ connection Slider2D::connect_submit_signal(
   return m_submit_signal.connect(slot);
 }
 
+bool Slider2D::event(QEvent* event) {
+  if(event->type() == QEvent::LayoutRequest) {
+    auto& track_stylist = find_stylist(*m_track);
```

Comment:
What is the relationship between a layout request and the styling? This looks very hacky.

Reply:
Yeah, you are correct. There is no relationship between them. If the QTimer is not allowed to used in the style signal handler, I cannot find a way to avoid calling match/unmatch within the same event as the style update, except by using QEvent::LayoutRequest or QEvent::Paint, which are emitted outside the style update event.